app/src/main/java/fr/free/nrw/commons/repository/UploadLocalDataSource.java

```diff
@@ -0,0 +1,60 @@
+package fr.free.nrw.commons.repository;
+
+import fr.free.nrw.commons.kvstore.BasicKvStore;
+import fr.free.nrw.commons.kvstore.JsonKvStore;
+import fr.free.nrw.commons.upload.UploadModel;
+import fr.free.nrw.commons.upload.UploadModel.UploadItem;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+@Singleton
+public class UploadLocalDataSource {
+
+    private final UploadModel uploadModel;
+    private BasicKvStore defaultKvStore;
+    private JsonKvStore directKvStore;
+
+    @Inject
+    public UploadLocalDataSource(@Named("default_preferences") BasicKvStore defaultKvStore,
```

Comment:
Ideally yes, but WikidataEditService pulls the value("Picture_Has_Correct_Location") from DirectKcStore. I could change it at both the places, but I thought it could be done in a separate PR. What do you suggest, shall I make the changes in this PR itself ?